### PR TITLE
Add scala scala submodule.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by DarkDimius** [comment]
LGTM after tests pass

**Comment by smarter** [comment]
I would call it scala-library, not scala-scala

File: compiler/test/dotty/tools/dotc/CompilationTests.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ class CompilationTests extends ParallelTesting {
 
   @Test def compilePos: Unit = {
     compileList("compileStdLib", StdLibSources.whitelisted, scala2Mode.and("-migration", "-Yno-inline")) +
+    compileDir("../collection-strawman/src/main", defaultOptions) +
```
Comment (DarkDimius) [inline compiler/test/dotty/tools/dotc/CompilationTests.scala]: maybe make it a separate test?

Comment (nicolasstucki) [inline compiler/test/dotty/tools/dotc/CompilationTests.scala]: We would loose the parallelism.

File: project/Build.scala [hunk 2]
```diff
@@ -857,12 +857,12 @@ object DottyInjectedPlugin extends AutoPlugin {
   }
 
   private def projectChecks(): Unit = {
-    val scalaScala = new File("scala-scala")
-    if (!scalaScala.exists()) {
+    val submodules = List(new File("scala-backend"), new File("scala-scala"), new File("collection-strawman"))
+    if (!submodules.forall(_.exists)) {
       println(
```
Comment (smarter) [inline project/Build.scala]: You shouldn't use println in the sbt build, see http://www.scala-sbt.org/0.13/docs/Howto-Logging.html#Log+messages+in+a+task for the proper way to do it. This should probably be logged at the error level.

**Comment by smarter** [comment]
Please go through the documentation `doc/doc/` to remove references to scala-scala.

File: project/Build.scala (hunk 2 above)
Comment (nicolasstucki) [inline project/Build.scala]: This is not a task.

Comment (smarter) [inline project/Build.scala]: But could it be one?

**Comment by nicolasstucki** [comment]
I noticed that if I set the `scala-library` submodule to the `lampepfl/scala` with branch `sharing-backend` the only test that fails is `dotty.tools.dottydoc.TestWhitelistedCollections.arrayAndImmutableHasDocumentation`. If we fix this we could remove use a single library/backend submodule. @DarkDimius @smarter what do you think about that?

**Comment by smarter** [comment]
@nicolasstucki You probably forgot to do `git submodule update`, there is no way that the library in sharing-backend compiles with dotty, it uses procedure syntax, etc.

**Comment by nicolasstucki** [comment]
@smarter indeed at first I forgot to do `git submodule update` and noticed that it passed all tests except `TestWhitelistedCollections` on `lampepfl/scala` with branch `master`. Then I explicitly change the branch to `sharing-backend` and saw the same. 

All the tests on the `scala-library` use the scala2 flag. That is why it can handle procedure syntax, etc.

**Comment by nicolasstucki** [comment]
@DarkDimius, could you check if it still LGTY. If this is merged it would simplify a bit the call graph branch tests.

**Comment by nicolasstucki** [comment]
@smarter, do you agree with the changes in the last commit?

**Comment by DarkDimius** [comment]
LGTM from my side, didn't check sbt though.

**Pull request merged**